scripts/sct_register_to_template.py

```diff
@@ -683,6 +690,52 @@ def generate_qc(fname_data, fname_template2anat, fname_seg, args, path_qc):
     )
 
 
+def project_labels_on_spinalcord(fname_label, fname_seg):
+    """
+    Project labels orthogonally on the spinal cord centerline. The algorithm works by finding the smallest distance
+    between each label and the spinal cord center of mass.
+    :param fname_label: file name of labels
+    :param fname_seg: file name of cord segmentation (could also be of centerline)
+    :return: file name of projected labels
+    """
+    # build output name
+    fname_label_projected = sct.add_suffix(fname_label, "_projected")
+    # open labels and segmentation
+    im_label = Image(fname_label)
+    im_seg = Image(fname_seg)
+    # orient to RPI
+    native_orient = im_seg.change_orientation('RPI')
+    im_label.change_orientation('RPI')
+    # smooth centerline and return fitted coordinates in voxel space
+    centerline_x, centerline_y, centerline_z, centerline_derivx, centerline_derivy, centerline_derivz = smooth_centerline(
+        im_seg, algo_fitting="hanning", type_window="hanning", window_length=50, nurbs_pts_number=3000,
+        phys_coordinates=False, all_slices=True)
+    # get center of mass of label
+    labels = im_label.getCoordinatesAveragedByValue()
+    # initialize image of projected labels
+    im_label_projected = im_label.copy()
+    im_label_projected.data = np.zeros(im_label_projected.data.shape, dtype='uint8')
+    # loop across label values
+    for label in labels:
+        # calculate distance between label and each point of the centerline
+        distance_centerline = [np.linalg.norm([centerline_x[i] - label.x,
+                                               centerline_y[i] - label.y,
+                                               centerline_z[i] - label.z]) for i in range(len(centerline_x))]
```

Comment:
If getCoordinatesAveragedByValue() and smooth_centerline() were returning arrays, the syntax could be prettier.

Reply:
agreed, another thing to add to the todo list :-)